flink-yarn/src/test/java/org/apache/flink/yarn/YarnClusterDescriptorTest.java

```diff
@@ -174,10 +174,26 @@ public void testConfigOverwrite() throws ClusterDeploymentException {
 	@Test
 	public void testSetupApplicationMasterContainer() {
 		Configuration cfg = new Configuration();
+		cfg.setBoolean(CoreOptions.FLINK_JVM_DEFAULT_GC_LOGGING, true);
 		YarnClusterDescriptor clusterDescriptor = createYarnClusterDescriptor(cfg);
 
 		final String java = "$JAVA_HOME/bin/java";
 		final String jvmmem = "-Xms424m -Xmx424m";
+		final String defaultGCLoggingOpts =
+			"-Xloggc:" + ApplicationConstants.LOG_DIR_EXPANSION_VAR + "/gc.log " +
+			"-XX:+PrintGCApplicationStoppedTime " +
+			"-XX:+PrintGCDetails " +
+			"-XX:+PrintGCDateStamps " +
+			"-XX:+UseGCLogFileRotation " +
+			"-XX:NumberOfGCLogFiles=10 " +
+			"-XX:GCLogFileSize=10M " +
+			"-XX:+PrintPromotionFailure " +
+			"-XX:+PrintGCCause";
+		final String heapdumpOpts =
+			"-XX:+HeapDumpOnOutOfMemoryError " +
+				"-XX:HeapDumpPath=/tmp/test-flink-jm-heapdump.hprof " +
+				"-XX:OnOutOfMemoryError=\"echo -e 'OutOfMemoryError! Killing current process %p...\nCheck gc logs and heapdump file(/tmp/test-flink-jm-heapdump.hprof) for details.' > " +
+				"<LOG_DIR>/jobmanager.err; kill -9 %p\"";
```

Comment:
Would be great to deduplicate this part as well. For example, if we have a utility `getGCLoggingOpts(....)`, then we could use it here.

Reply:
Thanks for your suggestion @tillrohrmann , I will try to deduplicate those code.

Reply:
Can we use the helper methods to generate the expected values instead of hard coding them here?